index.js

```diff
@@ -82,11 +76,13 @@ const factory = options => {
     opts.issWhitelist.indexOf(token.payload.iss) > -1 ||
     Promise.reject(new IssWhitelistError(`iss '${token.payload.iss}' not in issWhitelist`))
 
-  const checkExp = ({ payload }) =>
-    Promise.resolve(payload)
-      .then(res => res.exp)
-      .then(throwIfExpired(verifyOpts))
-      .catch(throwWithData({ payload }))
+  const checkExp = ({ ignoreExpiration }) => ({ payload: { exp } }) => {
+    if (ignoreExpiration) return
+
+    if (exp && exp > (Date.now() / 1000)) {
+      Promise.reject(new TokenExpiredError('JWT expired', new Date(exp * 999)))
```

Comment:
This function still has a mixed `Promise`/`undefined` contract.

```suggestion
      throw new TokenExpiredError('JWT expired', new Date(exp * 999))
```

At the very least, this `Promise.reject` needs returned.